addons/fleet/models/fleet_vehicle.py

```diff
@@ -259,12 +258,21 @@ def _search_get_overdue_contract_reminder(self, operator, value):
         else:
             search_operator = 'not in'
         today = fields.Date.context_today(self)
-        res_ids = self.env['fleet.vehicle.log.contract'].search([
+        # get the id of vehicles that have overdue contracts
+        expired_ids = self.env['fleet.vehicle.log.contract'].search([
             ('expiration_date', '!=', False),
             ('expiration_date', '<', today),
             ('state', 'in', ['open', 'expired'])
-        ]).mapped('vehicle_id').ids
-        res.append(('id', search_operator, res_ids))
+        ]).vehicle_id
+        # get the id of vehicles that have overdue contracts, but a new contract has already been created for them
+        resolved_ids = self.env['fleet.vehicle.log.contract'].search([
+            ('vehicle_id', 'in', expired_ids.ids),
+            ('expiration_date', '!=', False),
+            ('expiration_date', '>=', today),
+            ('state', 'in', ['open', 'futur'])
+        ]).vehicle_id
+        res_ids = expired_ids - resolved_ids
+        res.append(('id', search_operator, res_ids.ids))
```

Comment:
if you just need the ids for a domain you can just _search instead of search; the query will be delayed and the performances will be better.

Reply:
```
vehicle_ids = self.env['fleet.vehicle']._search([
            ("log_contracts", "any", [
                ('expiration_date', '!=', False),
                ('expiration_date', '<', today),
                ('state', 'in', ['open', 'expired'])
            ]),
            "!",
                ("log_contracts", "any", [
                    ('expiration_date', '!=', False),
                    ('expiration_date', '>=', today),
                    ('state', 'in', ['open', 'futur'])
                ]),
        ])
res.append(('id', search_operator, vehicle_ids))
```

Reply:
Well spotted ! :muscle: